spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDataSourceV2Implicits.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.connector.iceberg.catalog.SupportsMerge
+
+// must be merged with DataSourceV2Implicits in Spark
+object ExtendedDataSourceV2Implicits {
+  implicit class TableHelper(table: Table) {
+    def asMergeable: SupportsMerge = {
+      table match {
+        case support: SupportsMerge =>
+          support
+        case _ =>
+          throw new AnalysisException(s"Table does not support row level operations: ${table.name}")
```

Comment:
Currently this is just "supports row level merge and delete" so maybe we should detail that instead of "row level operations" which I feel like is a bit more vague.

Reply:
Done.